src/utils.ts

```diff
@@ -230,29 +231,34 @@ export async function retry<T>(call: () => Promise<T>, times: number, delayS: nu
  * @param unsignedTx The unsigned transaction that this function will estimate
  * @param senderAddress The address that the transaction will be submitted from
  * @param provider A valid ethers provider - will be used to reason the gas price
+ * @param gasMarkup Represents a percent increase on the total gas cost. For example, 0.2 will increase this resulting value by a factor of 1.2
  * @param gasPrice A manually provided gas price - if set, this function will not resolve the current gas price
  * @returns The total gas cost to submit this transaction - i.e. gasPrice * estimatedGasUnits
  */
 export async function estimateTotalGasRequiredByUnsignedTransaction(
   unsignedTx: PopulatedTransaction,
   senderAddress: string,
   provider: providers.Provider | L2Provider<providers.Provider>,
+  gasMarkup: number,
   gasPrice?: BigNumberish
 ): Promise<BigNumberish> {
+  assert(gasMarkup >= -1 && gasMarkup <= 4, "Gas Markup must be within the range of [-1.0, +4.0]");
```

Comment:
```suggestion
  assert(gasMarkup > -1 && gasMarkup <= 4, "Gas Markup must be within the range of (-1.0, +4.0] so that total gas multiplier is between (0, +5.0]");
```